addons/website/static/src/js/editor/wysiwyg.js

```diff
@@ -180,6 +184,31 @@ Wysiwyg.include({
 
         return ret;
     },
+    /**
+     * Hides all opened dropdowns.
+     *
+     * @private
+     */
+    _hideDropdowns() {
+        for (const toggleEl of this.el.querySelectorAll(
+            ".o_mega_menu_toggle, #top_menu_container .dropdown-toggle"
+        )) {
+            $(toggleEl).dropdown("hide");
+        }
+    },
+    /**
+     * Called when the page is clicked anywhere.
+     * Closes the shown dropdown if the click is outside of it.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onPageClick(ev) {
+        if (ev.target.closest(".dropdown.show")) {
+            return;
+        }
+        this._hideDropdowns();
+    },
```

Comment:
Also the `_hideDropdowns()` should be added in "Private" & `_onPageClick()` in "Handlers".